### PR TITLE
Revert to scala 2.12.9 and ignore MiMa false positives

### PR DESCRIPTION
**Pull request by regadas**
See for more context: https://github.com/lightbend/mima/issues/388

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -151,6 +151,56 @@ lazy val noPublishSettings = Seq(
   publishArtifact := false
 )
 
+// https://github.com/lightbend/mima/issues/388
+lazy val mimaSettings = Def.settings(
```
Comment (regadas) [inline build.sbt:155]: This is not pretty but I guess we will be able to remove this once that issue is fixed

**Comment by johnynek** [comment]
@regadas This seems fine to get things green for a release, but I think we should revert it after the next publish. What do you think?

**Comment by regadas** [comment]
@johnynek sounds good! 👍

**Comment by regadas** [comment]
flaky `tut`?

**Comment by johnynek** [comment]
I'll rerun but :`(run-main-0) java.lang.NoSuchMethodError: scala.reflect.internal.util.ReusableInstance.<init>(Lscala/Function0;)V` doesn't look flakey. It looks like binary incompatibility.

Maybe tut isn't compatible with one of these version bumps?

maybe an sbt caching bug?

**Comment by regadas** [comment]
Oh! I missed that! I'll have a look.

**Comment by johnynek** [comment]
Maybe we should revert the 2.12 bump. Master is red now and it caused all the scala steward bumps to fail.

**Comment by regadas** [comment]
Yup agreed! yesterday I tracked this a bit and I think there's something fishy around `tut` and `2.12.10` repl interpreter that it's instantiated; seems that the classpath is not right.

`2.12.9` is ok I'll push a commit here to double check.

**Review by nevillelyh — APPROVED** [review]

**Pull request merged**